### PR TITLE
Lucas/improve code and add tests

### PR DESCRIPTION
**Pull request by lgrossi**
Initially we were just validating concepts and possibilities and assessing the paths we wanted to follow.

Now we have a more solid idea of where to go and plans.

We are starting to design the code accordingly, adding tests and improving the architecture of the lib and components to fit our standards.

This PR aims to do a first clean up and modularization of code as well as introduce tests to functionalities added so far.

### CODE REVIEW

File: ryot/src/sprites/error.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Ryot - A free and open-source MMORPG server emulator
+ * Copyright (©) 2023 Lucas Grossi <lucas.ggrossi@gmail.com>
+ * Repository: https://github.com/lgrossi/Ryot
+ * License: https://github.com/lgrossi/Ryot/blob/main/LICENSE
+ * Contributors: https://github.com/lgrossi/Ryot/graphs/contributors
+ * Website: https://github.com/lgrossi/Ryot
+ */
```
Comment (luan) [inline ryot/src/sprites/error.rs:8]: The auto import feature on my editor keeps adding imports above this comment. Do we need to sign off every file with this?

File: ryot/src/sprites/error.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+pub type Result<T> = std::result::Result<T, Error>;
```
Comment (luan) [inline ryot/src/sprites/error.rs:9]: Check out [`color_eyre`](https://docs.rs/color-eyre/latest/color_eyre/) for a general error wrapper that provides a similar, but flexible and colorful `Result`

File: ryot/tests/appearances/mod.rs
Comment (luan) [inline ryot/tests/appearances/mod.rs:1]: These look like unit tests for the `appearances` module, right? According to [the book](https://doc.rust-lang.org/book/ch11-03-test-organization.html) I believe these would belong in `src/appearances/test.rs` (or inside the module itself).

File: ryot/src/sprites/error.rs (hunk 1 above)
Comment (lgrossi) [reply ryot/src/sprites/error.rs:8]: I think we can drop it and stop adding them

File: ryot/src/sprites/error.rs (hunk 2 above)
Comment (lgrossi) [reply ryot/src/sprites/error.rs:9]: Good point, I added it to my reading list last week and forgot to use it

File: ryot/tests/appearances/mod.rs
Comment (lgrossi) [reply ryot/tests/appearances/mod.rs:1]: Yeah, Im aware of that, I just really deslike the tests within the main classes. But I can move them back to keep consistency with rust standards

File: ryot/src/sprites/error.rs (hunk 2 above)
Comment (lgrossi) [reply ryot/src/sprites/error.rs:3]: Actually, I did move all of them to use this error, which simplifies a lot. However, I'll leave the usage of color_eyre for a next MR. It's slightly more complex than I thought, when I actually start refactoring the code. Then we can move all errors to reports aligned with color_eyre design.

File: ryot/tests/appearances/mod.rs
Comment (lgrossi) [reply ryot/tests/appearances/mod.rs:1]: Refactored it to be consistent with Rust Book.

**Pull request merged**